Firestore/Source/Local/FSTMemoryQueryCache.mm

```diff
@@ -72,112 +56,74 @@ - (instancetype)initWithPersistence:(FSTMemoryPersistence *)persistence {
 #pragma mark Query tracking
 
 - (TargetId)highestTargetID {
-  return _highestTargetID;
+  return _cache->highest_target_id();
 }
 
 - (ListenSequenceNumber)highestListenSequenceNumber {
-  return _highestListenSequenceNumber;
+  return _cache->highest_listen_sequence_number();
 }
 
 - (const SnapshotVersion &)lastRemoteSnapshotVersion {
-  return _lastRemoteSnapshotVersion;
+  return _cache->last_remote_snapshot_version();
 }
 
 - (void)setLastRemoteSnapshotVersion:(SnapshotVersion)snapshotVersion {
-  _lastRemoteSnapshotVersion = std::move(snapshotVersion);
+  _cache->set_last_remote_snapshot_version(snapshotVersion);
```

Comment:
Since in the iimplementation you are calling `last_remote_snapshot_version_ = std::move(version);` it is consistent to call with `_cache->set_last_remote_snapshot_version(std::move(snapshotVersion));` as well. Or you can fix the line below to remove `std::move`. Either calling `std::move` or not (consistently) for `SnapshotVersion` type is fine since that type is simple and easy to copy.

Reply:
Adding it for consistency, since it seems like someone wanted it at some point.